### DISCUSSION TITLE
Every ControlNet model works except for Reference_Only. Anyone else have a problem?

### DISCUSSION THREAD

**Discussion by Joeyd348039** [Mikubill/sd-webui-controlnet#1496] [Q&A]
Controlnet was working great for me up until a week ago. All the other Controlnet models work but when I go to use Reference Only it no longer works. I did have an LRU Cache error at first that I managed to fix. But when Reference Only goes to load it stays at 0% and then skips over to the Torch.Float16 VAE model. 

All the other models complete that first process and don't go over to the Torch.Float16 VAE model and work great.

**Comment by aaronkingdom** [comment]
same,me too

**Reply by aaronkingdom** [reply]
https://github.com/Mikubill/sd-webui-controlnet/discussions/1280#discussioncomment-6106651

**Reply by Joeyd348039** [reply]
No luck. Deleted those files and still have the problem.

**Comment by Deadstarrr** [comment]
What's the issue? I just installed controlnet for the first time and I wanted to test reference-only. My image looks nothing like the image I'm using. Did reference-only recently break? I was trying to figure out if I was doing something wrong. I've followed the tutorials online and the only thing I'm not sure of is what "Control Type" to pick.

**Reply by Joeyd348039** [reply]
It was about 2 weeks ago. Was working great and then it just stopped.  The only thing that I can guess is that the preprocessor isn't kicking in cause it get's skipped only on reference only.  Before it would create incredible images based off the same styling of the pic.

**Reply by Deadstarrr** [reply]
> It was about 2 weeks ago. Was working great and then it just stopped. The only thing that I can guess is that the preprocessor isn't kicking in cause it get's skipped only on reference only. Before it would create incredible images based off the same styling of the pic.

Thanks...because I was thinking I was doing something wrong. Without a prompt my image looks nothing like what I'm putting in. So we just need to wait for a controlnet fix?

**Reply by Joeyd348039** [reply]
I'm a bit confused. Reference Only is supposed to completely change the image. 

For me Reference Only has no impact on the image.

**Reply by Joeyd348039** [reply]
> > It was about 2 weeks ago. Was working great and then it just stopped. The only thing that I can guess is that the preprocessor isn't kicking in cause it get's skipped only on reference only. Before it would create incredible images based off the same styling of the pic.
> 
> Thanks...because I was thinking I was doing something wrong. Without a prompt my image looks nothing like what I'm putting in. So we just need to wait for a controlnet fix?

I did run the debug arg, I wonder if you would get the same results:

 DEBUG - Create LRU cache (max_size=16) for preprocessor results.
 DEBUG - Calling preprocessor reference_only outside of cache.

But I don't know much about how all this works so I'm not sure if these are bugs/errors that need fixing or not.

**Comment by lllyasviel** [comment]
Hello, please try the below steps and see if problems are solved:

1. update A1111 and CN to latest version. Make sure to restart terminal after update.
2. disable all other A1111 extensions.
3. disable all browser extensions or change another browser.
4. remove all cmd flags in A1111.
5. delete "config.json" and "ui-config.json". (dangerous, remember to backup files)
6. make sure that you have followed the official instruction to download ControlNet models, and make sure that each model is about 1.4 GB large.
7. uninstall ControlNet by removing the controlnet folder and try to install again.
8. make sure to not quit your webui when ControlNet is downloading preprocessor in the background terminal.
9. try to add “--no-gradio-queue” to your a1111 cmd flags.

The above steps can solve 99% problems. After your problem is solved, you can begin to enable other extensions one-by-one or add A1111 cmd flags one-by-one to find out what is causing problems and let us know.

If you are the unlucky 1%, you can try to use "git checkout" to find out if previous versions of CN works for you, and if that is the case, please let us know your last working version so that we can debug the problem.

If you still cannot solve the problem, please give as detailed as possible info about your environment.

**Reply by Joeyd348039** [reply]
I tried all that. Even rolled back my Nvidia driver in case that somehow was the problem.  I was trying to find the first version of ControlNet to download that had the reference only in it to see if that version works but no luck finding it.  And I accidentally deleted my old SD folder where I had all my original Controlnet images too , so I can't even see what day/time it all started :(

I've wiped out everything, done clean installs, no extensions, even disabled upscalers, etc..  I also set the preprocessor cache size to 16 in args commandline and no luck. Deleting configs and other stuff people have mentioned hasn't worked.

Reference Only stays at 0%. Every other model doesn't produce that error. 

I don't know for sure since I don't know programming but it seems like Reference Only isn't loading correctly so it doesn't get applied.

**Reply by lllyasviel** [reply]
can you share logs?

**Reply by Joeyd348039** [reply]
> can you share logs?

Is there a certain spot they are located at? Can't seem to find where they are.

**Reply by lllyasviel** [reply]
you can copy the content from terminal

**Reply by Joeyd348039** [reply]
> you can copy the content from terminal

Startup time: 7.0s (import torch: 1.1s, import gradio: 0.8s, import ldm: 0.4s, other imports: 0.6s, list SD models: 0.1s, load scripts: 1.0s, create ui: 2.9s).
2023-06-08 17:20:21,119 - ControlNet - INFO - Loading preprocessor: reference_only
2023-06-08 17:20:21,119 - ControlNet - INFO - Pixel Perfect Computation:
2023-06-08 17:20:21,120 - ControlNet - INFO - resize_mode = ResizeMode.INNER_FIT
2023-06-08 17:20:21,120 - ControlNet - INFO - raw_H = 768
2023-06-08 17:20:21,120 - ControlNet - INFO - raw_W = 1152
2023-06-08 17:20:21,120 - ControlNet - INFO - target_H = 512
2023-06-08 17:20:21,120 - ControlNet - INFO - target_W = 512
2023-06-08 17:20:21,120 - ControlNet - INFO - estimation = 512.0
2023-06-08 17:20:21,120 - ControlNet - INFO - preprocessor resolution = 512
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]ControlNet used torch.float16 VAE to encode torch.Size([1, 4, 64, 64]).
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:04<00:00,  4.52it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:03<00:00,  6.44it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:03<00:00,  6.27it/s]

-------

**Reply by lllyasviel** [reply]
I do not see any problem?

**Reply by Joeyd348039** [reply]
> I do not see any problem?

The first steps stay at 0%. 0/20 get completed. 

I need to see if I have any of my older Controlnet images stored anywhere. Reference only was incredible for styling, especially for img2img. Now it has zero effect.  

Not sure if this is an error but with debug on this pops up:
DEBUG - Calling preprocessor reference_only outside of cache.

**Reply by lllyasviel** [reply]
have you tried enable and disable CN and use same seed to see if it has some effects?

**Reply by Joeyd348039** [reply]
> have you tried enable and disable CN and use same seed to see if it has some effects?

Yeah. It's weird. It doesn't seem to really pick up on the styles at all. At most it may pick up on the structure of the image. Even with that it's barely noticeable most of the time. I was looking at videos on youtube and saw a few other comments of people having similar problems. It's like it's only working at 5% or so and picking up outlines instead of styles if that makes sense.

Before I could feed it just about any image and it would transfer styles with ease. I loved your Controlnet script, it was the only reason why I ended up using Stable Diffusion. 

I wanted to try to ask people or maybe even you if you get that 0% when you load up Reference Only too.

**Reply by lllyasviel** [reply]
closed as not bug - improvements are in future considerations
